### DISCUSSION TITLE
Get a list of changed components plus all their transitive dependants for testing

### DISCUSSION THREAD

**Discussion by jamesrobson-secondmind** [DavidVujic/python-polylith#236] [Ideas]
What I would like is a way to find any component that has changed and walk the tree of imports to find any component that could be affected by the change. This would let you test all code that might have been broken without needing to run everything. 
You can almost do this by combining `poly diff` and `poly deps` except that `poly deps` doesn't provide output in a format that's easy to parse.

**Comment by DavidVujic** [comment]
I like your idea a lot! Thank you @jamesrobson-secondmind 

I agree, it would be very useful to have this information in a parseable way. I'll think about how to implement it.

**Comment by DavidVujic** [comment]
Meanwhile, there are ways to get the changed bricks and use that data in an automated way. This can be used in combination with `pytest` queries and test markers: https://davidvujic.github.io/python-polylith-docs/testing/#running-tests-for-changed-code

This is about the actual changed bricks, and won't identify any other bricks using them. I'm looking into that part and will probably add something similar to the things already there (the link).

**Comment by DavidVujic** [comment]
@jamesrobson-secondmind I have released a new version that makes it possible to get data for dependent bricks. Announcement here: https://github.com/DavidVujic/python-polylith/discussions/244

**Comment by jamesrobson-secondmind** [comment]
Great, I'll give it a go.
